lib/buffers.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+var Buffers = require('buffers');
+
+Buffers.prototype.skip = function(i) {
+  if (i === 0) {
+    return;
+  }
+
+  if (i === this.length) {
+    this.buffers = [];
+    this.length = 0;
+    return;
+  }
+
+  var pos = this.pos(i);
+  this.buffers = this.buffers.slice(pos.buf);
+  this.buffers[0] = new Buffer(this.buffers[0].slice(pos.offset));
+  this.length -= i;
+};
+
+module.exports = Buffers;
+
```

Comment:
added test in: https://github.com/braydonf/bitcore-p2p/commit/7cfe6d18650d175499f75baa6b5cbe3c5ea46651